Typescript errors when building: `TS2578: Unused '@ts-expect-error' directive.`

### Where did this happen?

Local code with an up-to-date main branch

### Expected Behavior

No Typescript errors when building

### Current Behavior

```
yarn cli build nodePckgs -e staging

$ /Users/tyler/src/switchback/compass/node_modules/.bin/tsc --project tsconfig.build.json
packages/backend/src/auth/auth.routes.config.ts(23,7): error TS2578: Unused '@ts-expect-error' directive.
packages/backend/src/auth/auth.routes.config.ts(30,7): error TS2578: Unused '@ts-expect-error' directive.
packages/backend/src/auth/auth.routes.config.ts(35,9): error TS2578: Unused '@ts-expect-error' directive.
packages/backend/src/auth/auth.routes.config.ts(49,7): error TS2578: Unused '@ts-expect-error' directive.
packages/backend/src/common/middleware/promise.middleware.ts(33,5): error TS2322: Type '(p: Promise<any> | SyncFunction) => Promise<void>' is not assignable to type '(p: unknown) => Response<any, Record<string, any>, number>'.
  Types of parameters 'p' and 'p' are incompatible.
    Type 'unknown' is not assignable to type 'Promise<any> | SyncFunction'.
error Command failed with exit code 2.
```

### Steps to Reproduce
Run `yarn cli build nodePckgs -e staging`

### Possible Solution

Fix the Typescript issues in the app code

### Context

_No response_

--- a/packages/backend/src/auth/auth.routes.config.ts
+++ b/packages/backend/src/auth/auth.routes.config.ts
@@ -18,23 +18,16 @@ export class AuthRoutes extends CommonRoutesConfig {
     /**
      * Checks whether user's google access token is still valid
      */
-    this.app.route(`/api/auth/google`).get([
-      verifySession(),
-      //@ts-expect-error res.promise is not returning response types correctly
-      authController.verifyGToken,
-    ]);
+    this.app
+      .route(`/api/auth/google`)
+      .get([verifySession(), authController.verifyGToken]);
 
     this.app
       .route(`/api/auth/session`)
       .all(authMiddleware.verifyIsDev)
-      //@ts-expect-error res.promise is not returning response types correctly
       // eslint-disable-next-line @typescript-eslint/no-misused-promises
       .post(authController.createSession)
-      .get([
-        verifySession(),
-        //@ts-expect-error res.promise is not returning response types correctly
-        authController.getUserIdFromSession,
-      ]);
+      .get([verifySession(), authController.getUserIdFromSession]);
 
     this.app
       .route(`/api/auth/session/revoke`)
@@ -44,11 +37,12 @@ export class AuthRoutes extends CommonRoutesConfig {
     /**
      * Google calls this route after successful oauth
      */
-    this.app.route(`/api/oauth/google`).post([
-      authMiddleware.verifyGoogleOauthCode,
-      //@ts-expect-error res.promise is not returning response types correctly
-      authController.loginOrSignup,
-    ]);
+    this.app
+      .route(`/api/oauth/google`)
+      .post([
+        authMiddleware.verifyGoogleOauthCode,
+        authController.loginOrSignup,
+      ]);
 
     return this.app;
   }

--- a/packages/backend/src/common/middleware/promise.middleware.ts
+++ b/packages/backend/src/common/middleware/promise.middleware.ts
@@ -30,7 +30,7 @@ export const requestMiddleware = () => {
     res: Res_Promise,
     next: express.NextFunction,
   ) => {
-    res.promise = (p: Promise<any> | SyncFunction) => {
+    res.promise = (p: Promise<any> | SyncFunction | unknown) => {
       let toResolve: Promise<unknown> | (() => any);
 
       if (p instanceof Promise) {
@@ -41,9 +41,11 @@ export const requestMiddleware = () => {
         toResolve = Promise.resolve(p);
       }
 
-      return toResolve
+      toResolve
         .then((data) => sendResponse(res, data as D))
         .catch((e) => handleExpressError(res, e));
+
+      return res;
     };
 
     return next();

